### ISSUE TITLE
Try Question Pair similarity using Bert

### ISSUE THREAD

**Issue by Richik-main** [abhradeepd/NLP-Final-Project-Group-5#14]
I will try fine tuning two layers of a Bert architecture using the question pair similarity technique. I will pass two questions separated by a SEP token. And finally using BCEWithLogitsLoss for classification as this is a binary classification problem
![image](https://github.com/user-attachments/assets/a80167af-4d30-4563-b56c-0936b6994e1a)

**Comment by Richik-main** [comment]
As the data is imbalance 
![image](https://github.com/user-attachments/assets/bdb0dd25-a6b8-4efd-b895-c5fc90653ea7)
The BCELogitsLoss started overfitting. Thus I started experimenting with CrossEntropyLoss. 

- `CrossEntropyLoss` allows the model to learn separate logits for both classes, giving it more capacity to adapt to complex patterns in the data.
- The softmax function ensures better scaling of logits into probabilities.

**Comment by Richik-main** [comment]
As the data is imbalance 
![image](https://github.com/user-attachments/assets/bdb0dd25-a6b8-4efd-b895-c5fc90653ea7)
The BCELogitsLoss started overfitting. Thus I started experimenting with CrossEntropyLoss. 

`CrossEntropyLoss allows the model to learn separate logits for both classes, giving it more capacity to adapt to complex patterns in the data.
The softmax function ensures better scaling of logits into probabilities.
